spark/src/main/scala/geotrellis/spark/knn/KNearestMethods.scala

```diff
@@ -0,0 +1,66 @@
+package geotrellis.spark.knn
+
+import scala.collection.immutable.Map
+
+import geotrellis.util.MethodExtensions
+import geotrellis.vector.{Extent, Point, Geometry, Feature}
+
+import org.apache.spark.rdd.RDD
+
+trait KNearestMethods[T] extends MethodExtensions[RDD[T]] {
+  def kNearest(x: Double, y: Double, k: Int)(f: T => Extent): Seq[T] = { KNearestRDD.kNearest[T](self, x, y, k)(f) }
+
+  def kNearest(p: (Double, Double), k: Int)(f: T => Extent): Seq[T] = { KNearestRDD.kNearest[T](self, p, k)(f) }
+
+  def kNearest(p: Point, k: Int)(f: T => Extent): Seq[T] = { KNearestRDD.kNearest[T](self, p, k)(f) }
+
+  def kNearest(ex: Extent, k: Int)(f: T => Extent): Seq[T] = { KNearestRDD.kNearest[T](self, ex, k)(f) }
+
+  def kNearest[H <: Geometry](centers: Traversable[H], k: Int)(f: T => Extent): Seq[Seq[T]] = {
+    KNearestRDD.kNearest[T, H](self, centers, k)(f, _.envelope)
+  }
+
+  def kNearest[H <: Geometry, F](centers: Traversable[Feature[H, F]], k: Int)(f: T => Extent)(implicit d: DummyImplicit): Seq[Seq[T]] = {
+    KNearestRDD.kNearest[T, Feature[H, F]](self, centers, k)(f, _.geom.envelope)
+  }
+}
+
+trait KNearestGeometryMethods[G <: Geometry] extends MethodExtensions[RDD[G]] {
+  def kNearest(x: Double, y: Double, k: Int): Seq[G] = KNearestRDD.kNearest[G](self, x, y, k){ g: G => g.envelope }
+
+  def kNearest(p: (Double, Double), k: Int): Seq[G] = KNearestRDD.kNearest[G](self, p, k){ g: G => g.envelope }
+
+  def kNearest(p: Point, k: Int): Seq[G] = KNearestRDD.kNearest[G](self, p, k){ g: G => g.envelope }
+
+  def kNearest(ex: Extent, k: Int): Seq[G] = KNearestRDD.kNearest[G](self, ex, k){ g: G => g.envelope }
+
+  def kNearest[H <: Geometry](centers: Traversable[H], k: Int): Seq[Seq[G]] = {
+    KNearestRDD.kNearest[G, H](self, centers, k)(_.envelope, _.envelope)
+  }
+
+  def kNearest[H <: Geometry, F](centers: Traversable[Feature[H, F]], k: Int)(implicit d: DummyImplicit): Seq[Seq[G]] = {
+    KNearestRDD.kNearest[G, Feature[H, F]](self, centers, k)(_.envelope, _.geom.envelope)
+  }
+}
+
+trait KNearestFeatureMethods[G <: Geometry, D] extends MethodExtensions[RDD[Feature[G, D]]] {
+  def kNearest(x: Double, y: Double, k: Int): Seq[Feature[G, D]] = 
+    KNearestRDD.kNearest[Feature[G, D]](self, x, y, k){ g: Feature[G, D] => g.geom.envelope }
+
+  def kNearest(p: (Double, Double), k: Int): Seq[Feature[G, D]] = 
+    KNearestRDD.kNearest[Feature[G, D]](self, p, k){ g: Feature[G, D] => g.geom.envelope }
+
+  def kNearest(p: Point, k: Int): Seq[Feature[G, D]] = 
+    KNearestRDD.kNearest[Feature[G, D]](self, p, k){ g: Feature[G, D] => g.geom.envelope }
+
+  def kNearest(ex: Extent, k: Int): Seq[Feature[G, D]] = 
+    KNearestRDD.kNearest[Feature[G, D]](self, ex, k){ g: Feature[G, D] => g.geom.envelope }
+
+  def kNearest[H <: Geometry](centers: Traversable[H], k: Int): Seq[Seq[Feature[G, D]]] = {
```

Comment:
This does not end up using the geometry's `distance` method, instead using the extent's, correct?

This will produce bad results, e.g. squiggly polygons with large bounding boxes.